src/Features/LanguageServer/ProtocolUnitTests/Hover/HoverTests.cs

```diff
@@ -428,6 +428,34 @@ void A.AMethod(int i)
             Assert.Equal(expectedMarkdown, results.Contents.Value.Fourth.Value);
         }
 
+        [Theory, CombinatorialData, WorkItem("https://github.com/dotnet/vscode-csharp/issues/6577")]
+        public async Task TestGetHoverAsync_UsesInlineCodeFencesInAwaitReturn(bool mutatingLspWorkspace)
+        {
+            var markup =
+@"using System.Threading.Tasks;
+class C
+{
+    public async Task<string> DoAsync()
+    {
+        return {|caret:await|} DoAsync();
+    }
+}";
+            var clientCapabilities = new LSP.ClientCapabilities
+            {
+                TextDocument = new LSP.TextDocumentClientCapabilities { Hover = new LSP.HoverSetting { ContentFormat = [LSP.MarkupKind.Markdown] } }
+            };
+            await using var testLspServer = await CreateTestLspServerAsync(markup, mutatingLspWorkspace, clientCapabilities);
+            var expectedLocation = testLspServer.GetLocations("caret").Single();
+
+            var expectedMarkdown = @"Awaited task returns '`class System.String`'  
```

Comment:
Where are the outer single quotes coming from? If we can mark the type as a code, I'm not sure we need those? It looked a bit funny to me.

Reply:
This is coming from the resource string, and is the same as VS
<img width="335" alt="image" src="https://github.com/dotnet/roslyn/assets/5749229/3885165f-3cf2-468e-8e49-9dab77eda8f8">

Reply:
Removing them might look a little weird in VS - as it is hard to distinguish between the text and the code without them.

Reply:
I guess I'm OK with removing on the VS side, since we've already got classification to help out too. I'm not sure if we do this more generally though and this is consistent.

Reply:
I'm fine with removing them - its a relatively simple change